TACC/Core-Styles
# feat(components): c-form & django-cms-forms, now wip status

## Overview

Promote Form components to "Work In Progress" status.

## Related

- [TUP-369](https://jira.tacc.utexas.edu/browse/TUP-369)
- [FP-1178](https://jira.tacc.utexas.edu/browse/FP-1178)
- required by https://github.com/TACC/Core-CMS/pull/569
- required by https://github.com/TACC/tup-ui/pull/70
- builds atop #77

## Changes

- changed load of pattern scripts to require modules
- changed error selectors to be simpler
- fixed required labels to **not** be bold
- changed error text to **not** be small
- added error styles
- added style for obvious buttons that neglect class
- changed errors to use `<ul>` not `<div>`
- removed Django CMS Form idiosyncrasies from C Form markup
- changed "fish" field to not be required
    <sup>multi-checkbox fields cannot be required</sup>
- added buttons to toggle errors and fields being required
- removed prototype from form patterns
- added scripts to toggle errors and fields being required
- added distinct styles for C Message scopes
- added link styles for C Message
- added sample errors to Django CMS Form markup
- added C Form to `core-styles.base.css`
- fixed inconsistent height of input and select fields
- added RGB colors
- fixed C Message colors to be accessible

## Testing

0. Build and serve demo.
1. Open C Form and Django CMS Forms components.
2. Verify both look identical.
3. Verify toggle buttons work:
    | Button | Action |
    | - | - |
    | Toggle Errors | Will show or hide form and field error messages. |
    | Toggle Required Fields | Will require or un-require fields that load as required. |

## UI

https://user-images.githubusercontent.com/62723358/205219719-b9c8471a-be77-405a-bb94-1d7d8e258419.mov

https://user-images.githubusercontent.com/62723358/205219705-2cb45af6-fb52-493c-b2a3-54fb35f809d7.mov

## Notes

The "Django CMS Form" belongs **only** in https://github.com/TACC/Core-CMS, but it is in Core Styles to expedite cross-pattern testing until both are "Ready" (not "Work in Progress") which requires more client testing.
